code/migration/src/updateUserRoles.js

```diff
@@ -0,0 +1,153 @@
+    .catch((err) => { throw err; });
+
+const getRoles = user =>
+  getAuthorisationToken()
+    .then(token =>
+      axios.get(`${authZeroAuthApi}/users/${user.userId}/roles`, createHeaders(token)))
+    .then(response => get(response, 'data'))
+    .then(roleList => roleList.map(role => role.name))
+    .then(roles => ({ ...user, roles }))
+    .catch((err) => { throw err; });
+
+const getUsersRoles = users =>
+  Promise.mapSeries(users, user => getRoles(user));
+
+const getKnownExpectedRoles = name => get(find(knownUserRoles, { name }), 'roles', []);
+
+const knownUsers = knownUserRoles.map(knownUser => knownUser.name);
+
+const addExpectedRoles = users =>
+  users.map((user) => {
+    if (knownUsers.findIndex(knownUser => knownUser === user.name) > -1) {
+      return { ...user, expectedRoles: getKnownExpectedRoles(user.name) };
+    }
+    return { ...user, expectedRoles: unknownUserRoles };
+  });
+
+const findRoleDiff = users =>
+  users.map(user => ({
+    ...user,
+    addRoles: difference(user.expectedRoles, user.roles).map(roleName => ({ roleName })),
+    removeRoles: difference(user.roles, user.expectedRoles).map(roleName => ({ roleName })),
+  }));
+
+const selectModifiedUsers = users =>
+  users.filter(({ addRoles, removeRoles }) => addRoles.length > 0 || removeRoles.length > 0);
+
+const getRoleId = (name, roleList) => find(roleList, { name });
+
+const findRoleIds = users =>
+  getAllRoles()
+    .then(roleList =>
+      users.map(user => ({
+        ...user,
+        addRole: user.addRoles.map(({ roleName }) => getRoleId(roleName, roleList)),
+        removeRole: user.removeRoles.map(({ roleName }) => getRoleId(roleName, roleList)),
+      })));
+
+const addUserRoles = user =>
+  getAuthorisationToken()
+    .then(token =>
+      axios.patch(`${authZeroAuthApi}/users/${user.userId}/roles`, [user.role.id], createHeaders(token)))
+    .then(() => logger.info(`Added user "${user.name}" to role "${user.role.name}"`))
+    .then(() => user);
+
+const execAddUserRoles = users =>
+  Promise.mapSeries(users, user =>
+    Promise.mapSeries(user.addRole, role => addUserRoles({ ...user, role })))
+    .then(() => users);
+
+const removeUserRoles = user =>
+  getAuthorisationToken()
+    .then(token =>
+      axios.delete(`${authZeroAuthApi}/users/${user.userId}/roles`, {
+        data: [user.role.id],
+        ...createHeaders(token),
+      }))
+    .then(() => logger.info(`Remove user "${user.name}" from role "${user.role.name}"`))
+    .then(() => user);
+
+const execRemoveUserRoles = users =>
+  Promise.mapSeries(users, user =>
+    Promise.mapSeries(user.removeRole, role => removeUserRoles({ ...user, role })))
+    .then(() => users);
+
+const updateUsers = () =>
+  getUserList()
+    .then(getUsersRoles)
+    .then(addExpectedRoles)
+    .then(findRoleDiff)
+    .then(selectModifiedUsers)
+    .then(findRoleIds)
+    .then(execAddUserRoles)
+    .then(execRemoveUserRoles)
+    .then(() => 'done');
```

Comment:
This doesn't log does it? Where are you returning `done` to?

Reply:
This is logged in `main.js`